de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/negotiation/AbstractIncomingProjectNegotiation.java

```diff
@@ -174,6 +147,31 @@ public Status run(Map<String, IProject> projectMapping, final IProgressMonitor m
     return terminate(exception);
   }
 
+  /**
+   * Adds all projects and resources to the session. This notifies all listener about the completed
```

Comment:
I cannot spot any listener call here.

Reply:
listeners of `resourcesAdded` in `session.addSharedResources(...)`

Reply:
I am aware of the broadcasting. However you have no control of what is broadcasted in this method call. In this case you can argue that the broadcast is a result of a finished negotiation. I still find the comment confusing as there is no listener invoked which actually announce a completed negotiation. This is in fact done when you call the terminate method.

And btw, addProjectsToSession is also not correct. It adds resources which may be projects but in case of partial sharing it can really only be resources that are added to an already shared project.

Reply:
removed the listener comment and renamed it